cli/tests/upgrade.bats

```diff
@@ -232,3 +232,21 @@ EOF
   assert_success
   assert_output "⬆️  Upgraded 'hello' in environment 'test'."
 }
+
+# bats test_tags=upgrade:flake:iid
+@test "upgrade for flake installable by iid" {
+  "$FLOX_BIN" init
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+  version = 1
+
+  [install]
+  hello.flake = "github:NixOS/nixpkgs#hello"
+EOF
+  )"
+
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
```

Comment:
This and the preceding test take 27~46s each on GitHub Runners:

- https://github.com/flox/flox/actions/runs/12280719379/job/34267823039#step:8:332
- https://github.com/flox/flox/actions/runs/12280719379/job/34267825532#step:8:331

We could save a lot of that by pinning to the same nixpkgs rev that we use [elsewhere](https://github.com/flox/flox/blob/64336892f12c16586fda00312a4f98eaa9273ca1/cli/tests/install.bats#L465-L481):
```suggestion
  run "$FLOX_BIN" install "github:nixos/nixpkgs/$PKGDB_NIXPKGS_REV_NEW#hello"
```

Reply:
ouff yeah, they should be pinned indeed